Maple2.Model/Game/InteractObject.cs

```diff
@@ -16,19 +16,24 @@ public abstract class InteractObject<T> : IInteractObject where T : InteractObje
     public abstract InteractType Type { get; }
 
     protected T Metadata { get; init; }
-    public int Id {get; init; }
-    public string Model = string.Empty;
-    public string Asset = string.Empty;
-    public string NormalState = string.Empty;
-    public string Reactable = string.Empty;
-    public float Scale = 1f;
+    public int Id { get; init; }
+    public string Model { get; init; }
+    public string Asset { get; init; }
+    public string NormalState { get; init; }
+    public string Reactable { get; init; }
+    public float Scale { get; init; }
 
-    public string EntityId { get; }
+    public string EntityId { get; init; }
 
-    protected InteractObject(string entityId, T metadata) {
+    protected InteractObject(string entityId, T metadata, string model = "", string asset = "", string normalState = "", string reactable = "", float scale = 1f) {
```

Comment:
do you need these parameters? the whole point of init params is that you don't need to pass it in the constructor. Just set the default to the param:

```cs
public string Param { get; init; } = "";
```